odbc/tests/connection_tests.rs

```diff
@@ -1,5 +1,17 @@
+use std::sync::Mutex;
+
 mod common;
 
+static TEST_SYNC: Mutex<Executor> = Mutex::new(Executor);
+#[derive(Clone, Copy)]
+struct Executor;
+
+impl Executor {
+    fn run_test(self, f: impl FnOnce()) {
+        f()
+    }
+}
+
```

Comment:
Olympics! This essentially forces `cargo test -- --test-threads=1`. We do this in the evergreen file, but this is nice when you just want to run `cargo test` from the command line.